kolibri/core/assets/src/mixins/responsive-element.js

```diff
@@ -0,0 +1,49 @@
+
+/*
+  Apply this mixin to your vue components to get reactive information
+  about the component's size.
+
+  For example:
+
+    <script>
+
+      const responsiveElement = require('./responsive-element-mixin.js')
+
+      export default {
+        mixins: [responsiveElement],
+        props: {
+    ...
+
+  This adds a new reactive property called `elSize` to your vue model:
+
+    elSize: {
+      width: 0,    // component's $el width (px)
+      height: 0,   // component's $el height (px)
+    },
+*/
+
+
+const ResizeSensor = require('css-element-queries/src/ResizeSensor');
+
+module.exports = {
+  data() {
+    return {
+      // becomes available for use
+      elSize: { width: 0, height: 0 },
+    };
+  },
+  methods: {
+    _updateEl() {
+      this.elSize.width = this.$el.clientWidth;
+      this.elSize.height = this.$el.clientHeight;
+    },
+  },
+  mounted() {
```

Comment:
yup

> hook functions with the same name are merged into an array so that all of them will be called. In addition, mixin hooks will be called before the component’s own hooks